mdn/developer-portal
# 1451: Frontend for Posts search

This changeset adds an updated UI for the Posts page, so that search parameters can be added via the filter form:

![Screenshot 2020-06-08 at 13 07 00](https://user-images.githubusercontent.com/101457/84033881-76ff0d00-a991-11ea-9fba-68213ec9468c.png)

![Screenshot 2020-06-08 at 12 56 16](https://user-images.githubusercontent.com/101457/84033937-8716ec80-a991-11ea-8dd6-29a7a9f91b40.png)
![Screenshot 2020-06-08 at 13 06 42](https://user-images.githubusercontent.com/101457/84033940-88481980-a991-11ea-8b16-422b38016176.png)
![Screenshot 2020-06-08 at 13 36 56](https://user-images.githubusercontent.com/101457/84033942-88e0b000-a991-11ea-8311-1f18afdcf138.png)
![Screenshot 2020-06-08 at 13 41 44](https://user-images.githubusercontent.com/101457/84033948-89794680-a991-11ea-9ccb-d27809618c0e.png)

(Related issue #1451)

## Key changes:

- search input added, with appropriate sizing for desktop and mobile breakpoints (inlcuding Safari vs mobile Chrome and FF)
- JS updated to handle clearing the search input, showing the "clear" trigger appropriately, etc  
- backend code updated slightly to further sanitise input string - see inline comment in PR 
- cross-browser tested

## How to test

- Code is staged on [Staging](https://developer-portal.stage.mdn.mozit.cloud/posts/) so can be tested without a local build
- Please try any/all search params you might normally try, as well as injection attacks, bad input etc - try to break it!
- confirm pagination works ok in concert with search and filters (eg search for 'CSS' while also having the 'CSS' filter enabled and confirm the URLs for each page are consistent and correct
- confirm the filtering on /events/ and /communities/posts/ still works as normal
- feel free to cross-browser test on real devices, too!
